test/common/performance/receiver/receiver.go

```diff
@@ -67,10 +66,14 @@ func NewReceiver(paceFlag string, aggregAddr string, typeExtractor TypeExtractor
 
 	// Calculate timeout for receiver
 	var timeout time.Duration
+	timeout = time.Second * time.Duration(warmupSeconds)
+	if timeout != 0 {
+		timeout += common.WaitAfterWarmup
+	}
 	for _, p := range pace {
 		timeout += p.Duration + common.WaitForFlush + common.WaitForReceiverGC
 	}
-	timeout += timeoutAdditionalTime
+	timeout *= 2 // Let's double it
```

Comment:
Why did this change?

Reply:
Tldr: this is because the sender is slowed down by the SUT when the load is too high and test requires more than needed

When the load is too high for the SUT, the sender experiences a "cap" and it's slowed down.
I'm not really sure if the cap is because of too much goroutines or too much HTTP connections waiting for a response, what matters is that the test could take additional time than required. This is the so called Coordinated Omission Problem (https://github.com/slinkydeveloper/eventing-bench-results/blob/master/11-14-2019/README.md#results)

That means that even if the SUT continues to reply, but slowly, the sender could be capped and good part of messages is never received because the receiver goes timeout, even if they were enqueued waiting to be processed.  
Based on experience, I've found that `timeout *= 2` is a good value (I've tried also `timeout *= 1.5` and `timeout *= 1 + 2/3`). Moreover the `timeoutAdditionalTime` was wrong because it was not proportional to the effective test time, that influences the queue waiting time